src/sync/impl/sync_file.cpp

```diff
@@ -180,20 +193,54 @@ std::string file_path_by_appending_extension(const std::string& path, const std:
     return buffer;
 }
 
+std::string create_timestamped_template(const std::string& prefix, int wildcard_count, util::Optional<std::time_t> time)
+{
+    constexpr int WILDCARD_MAX = 20;
+    constexpr int WILDCARD_MIN = 6;
+    // Clamp
+    wildcard_count = (wildcard_count > WILDCARD_MAX ? WILDCARD_MAX : wildcard_count);
+    wildcard_count = (wildcard_count < WILDCARD_MIN ? WILDCARD_MIN : wildcard_count);
+    std::time_t actual_time = time.value_or(std::time(nullptr));
+    std::stringstream stream;
+    stream << prefix << "-" << std::put_time(std::localtime(&actual_time), "%Y%m%d-%H%M%S") << "-";
+    for (int i = 0; i < wildcard_count; i++)
+        stream << "X";
```

Comment:
You could use `std::string(size_t count, char ch)` constructor that constructs a string with `count` copies of `ch`.